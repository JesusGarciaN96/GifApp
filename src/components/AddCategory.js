import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddCategory({ addCategory }) {
  const [category, setCategory] = useState('');

  const handleChangesCategory = ({ target }) => setCategory(target.value);

  const handleSubmitCategory = event => {
    event.preventDefault();
    if (category) {
      addCategory(cat => [...cat, category]);
      setCategory('');
    }
  };

  return (
    <form onSubmit={handleSubmitCategory}>
      <input
        type='text'
        value={category}
        onChange={handleChangesCategory}
        placeholder='Agregar Categoria'
      />
      <button type='submit'>Agregar Categoria</button>
    </form>
  );
}

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};

export default AddCategory;
