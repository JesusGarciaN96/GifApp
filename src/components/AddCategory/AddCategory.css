.add-category {
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.add-category__search-box {
  width: 90%;
}

.add-category__button {
  height: 34px;
  font-size: 14px;
  background-color: mediumpurple;
  color: white;
  border: none;
  border-radius: 3px;
  cursor: pointer;
  box-shadow: 1px 2px 3px rgba(0, 0, 0, 0.5);
}
