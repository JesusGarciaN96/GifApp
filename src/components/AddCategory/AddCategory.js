import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './AddCategory.css';

function AddCategory({ addCategory }) {
  const [category, setCategory] = useState('');

  const handleChangesCategory = ({ target }) => setCategory(target.value);

  const handleSubmitCategory = event => {
    event.preventDefault();
    if (category) {
      addCategory(cat => [category, ...cat]);
      setCategory('');
    }
  };

  return (
    <>
      <form className='add-category' onSubmit={handleSubmitCategory}>
        <input
          className='add-category__search-box'
          type='text'
          value={category}
          onChange={handleChangesCategory}
          placeholder='Agregar Categoria'
        />
        <button className='add-category__button' type='submit'>
          Agregar Categoria
        </button>
      </form>
      <p className='add-category__test-paragraph'>{category}</p>
    </>
  );
}

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};

export default AddCategory;
