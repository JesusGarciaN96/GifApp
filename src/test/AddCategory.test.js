import React from 'react';
import { shallow } from 'enzyme';

import AddCategory from '../components/AddCategory/AddCategory';

describe('Pruebas unitarias para componente AddCategory', () => {
  const agregarCategoria = () => [
    'One Punch Man',
    'Dragon Ball Super',
    'Pokemon',
  ];
  const wrapper = shallow(<AddCategory addCategory={agregarCategoria} />);

  test('Render del componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // El segundo parametro de 'simulate' son las propiedades, simula
  // el 'e' o 'event' -> 'event.target.value'
  test('Debe detectar cambios en la caja de busqueda', () => {
    const cajaBusqueda = wrapper.find('input');
    const valorSimulacion = 'Yu gi Oh!';
    cajaBusqueda.simulate('change', {
      target: {
        value: valorSimulacion,
      },
    });
    const textoBusqueda = wrapper.find('p').text().trim();
    expect(textoBusqueda).toBe(valorSimulacion);
  });
});
