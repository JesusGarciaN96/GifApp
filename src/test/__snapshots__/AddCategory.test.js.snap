// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias para componente AddCategory Render del componente 1`] = `
<Fragment>
  <form
    className="add-category"
    onSubmit={[Function]}
  >
    <input
      className="add-category__search-box"
      onChange={[Function]}
      placeholder="Agregar Categoria"
      type="text"
      value=""
    />
    <button
      className="add-category__button"
      type="submit"
    >
      Agregar Categoria
    </button>
  </form>
  <p
    className="add-category__test-paragraph"
  />
</Fragment>
`;
